src/uucore/src/lib/features/checksum.rs

```diff
@@ -440,41 +518,40 @@ fn get_filename_for_output(filename: &OsStr, input_is_stdin: bool) -> String {
     .to_string()
 }
 
-/// Determines the appropriate regular expression to use based on the provided lines.
-fn determine_regex(lines: &[OsString]) -> Option<(Regex, bool)> {
-    let regexes = [
-        (Regex::new(ALGO_BASED_REGEX).unwrap(), true),
-        (Regex::new(DOUBLE_SPACE_REGEX).unwrap(), false),
-        (Regex::new(SINGLE_SPACE_REGEX).unwrap(), false),
-        (Regex::new(ALGO_BASED_REGEX_BASE64).unwrap(), true),
-    ];
-
-    for line in lines {
-        let line_bytes = os_str_as_bytes(line).expect("UTF-8 decoding failed");
-        for (regex, is_algo_based) in &regexes {
-            if regex.is_match(line_bytes) {
-                return Some((regex.clone(), *is_algo_based));
-            }
-        }
-    }
-
-    None
-}
-
 /// Extract the expected digest from the checksum string
-fn get_expected_digest_as_hex_string(caps: &Captures, chosen_regex: &Regex) -> Option<String> {
-    // Unwraps are safe, ensured by regex.
-    let ck = caps.name("checksum").unwrap().as_bytes();
-
-    if chosen_regex.as_str() == ALGO_BASED_REGEX_BASE64 {
-        BASE64.decode(ck).map(hex::encode).ok()
-    } else if ck.len() % 2 == 0 {
-        Some(str::from_utf8(ck).unwrap().to_string())
-    } else {
+fn get_expected_digest_as_hex_string(
+    line_info: &LineInfo,
+    len_hint: Option<usize>,
+) -> Option<Cow<str>> {
+    let ck = &line_info.checksum;
+    let against_hint = |len| len_hint.is_some_and(|l| l != len);
```

Comment:
This closure and it's usage with the negation doesn't feel intuitive to me. I think it would be more intuitive with `is_none_or`. As this function is only available since Rust 1.82, we would have to provide our own implementation, though.
```suggestion
    let against_hint = |len| len_hint.is_none_or(|l| l == len);
```
And somewhere else in the file:
```rust
// TODO: remove once we have an MSRV of 1.82
// Add the is_none_or method to Option
#[allow(dead_code)]
trait OptionExt<T> {
    fn is_none_or<F>(self, f: F) -> bool
    where
        F: FnOnce(T) -> bool;
}

impl<T> OptionExt<T> for Option<T> {
    fn is_none_or<F>(self, f: F) -> bool
    where
        F: FnOnce(T) -> bool,
    {
        match self {
            Some(value) => f(value),
            None => true,
        }
    }
}

```

Reply:
I agree that it is counter intuitive, however I'm a bit reluctant to duplicate yet another rust-std feature to the project.
How about we add a TODO for when MSRV reaches 1.82 ?

Speaking about MSRV, is there a specific reason for which we avoid bumping it so much ?

Reply:
Yes, adding a TODO is also fine.

> Speaking about MSRV, is there a specific reason for which we avoid bumping it so much ?

I don't know if there is a specific reason, maybe @sylvestre can say more about it?